packages/automerge-repo/src/DocHandle.ts

```diff
@@ -78,156 +63,93 @@ export class DocHandle<T> //
 
     this.#log = debug(`automerge-repo:dochandle:${this.documentId.slice(0, 5)}`)
 
-    /**
-     * Internally we use a state machine to orchestrate document loading and/or syncing, in order to
-     * avoid requesting data we already have, or surfacing intermediate values to the consumer.
-     *
-     *                          ┌─────────────────────┬─────────TIMEOUT────►┌─────────────┐
-     *                      ┌───┴─────┐           ┌───┴────────┐            │ unavailable │
-     *  ┌───────┐  ┌──FIND──┤ loading ├─REQUEST──►│ requesting ├─UPDATE──┐  └─────────────┘
-     *  │ idle  ├──┤        └───┬─────┘           └────────────┘         │
-     *  └───────┘  │            │                                        └─►┌────────┐
-     *             │            └───────LOAD───────────────────────────────►│ ready  │
-     *             └──CREATE───────────────────────────────────────────────►└────────┘
-     */
-    this.#machine = interpret(
-      createMachine<DocHandleContext<T>, DocHandleEvent<T>>(
-        {
-          predictableActionArguments: true,
-
-          id: "docHandle",
-          initial: IDLE,
-          context: { documentId: this.documentId, doc },
-          states: {
-            idle: {
-              on: {
-                // If we're creating a new document, we don't need to load anything
-                CREATE: { target: READY },
-                // If we're accessing an existing document, we need to request it from storage
-                // and/or the network
-                FIND: { target: LOADING },
-                DELETE: { actions: "onDelete", target: DELETED },
```

Comment:
The UPDATE and DELETE event handlers were defined repeatedly throughout the config; since they apply the same way to every state, we can just define them once each at the root.